src/Onboarding/ExplanationScreen.tsx

```diff
@@ -1,197 +1,150 @@
 import React, { FunctionComponent } from "react"
 import {
-  ImageBackground,
-  TouchableOpacity,
+  Image,
   StyleSheet,
-  TextStyle,
   View,
   ScrollView,
   ImageSourcePropType,
   ViewStyle,
+  TouchableOpacity,
+  SafeAreaView,
 } from "react-native"
-import { SvgXml } from "react-native-svg"
+import { useNavigation } from "@react-navigation/native"
+import { useTranslation } from "react-i18next"
 
 import { Button } from "../components/Button"
-import { useStatusBarEffect, StatusBarStyle } from "../navigation"
 import { GlobalText } from "../components/GlobalText"
+import { OnboardingScreens } from "../navigation"
 
-import {
-  Outlines,
-  Buttons,
-  Colors,
-  Spacing,
-  Iconography,
-  Typography,
-} from "../styles"
-
-export enum IconStyle {
-  Blue,
-  Gold,
-}
+import { Layout, Outlines, Colors, Spacing, Typography } from "../styles"
 
 type ExplanationScreenContent = {
-  icon: string
-  iconLabel: string
+  screenNumber: number
+  image: ImageSourcePropType
+  imageLabel: string
   header: string
-  body: string
   primaryButtonLabel: string
-  backgroundImage: ImageSourcePropType
-  secondaryButtonLabel?: string
-}
-
-type ExplanationScreenStyles = {
-  headerStyle?: TextStyle
-  bodyStyle?: TextStyle
-  primaryButtonContainerStyle?: ViewStyle
-  primaryButtonTextStyle?: TextStyle
-  secondaryButtonContainerStyle?: ViewStyle
-  secondaryButtonTextStyle?: TextStyle
-  backgroundStyle?: ViewStyle
-  iconStyle: IconStyle
-  statusBarStyle: StatusBarStyle
 }
 
 type ExplanationScreenActions = {
   primaryButtonOnPress: () => void
-  secondaryButtonOnPress?: () => void
 }
 
 interface ExplanationScreenProps {
   explanationScreenContent: ExplanationScreenContent
-  explanationScreenStyles: ExplanationScreenStyles
   explanationScreenActions: ExplanationScreenActions
 }
 
 const ExplanationScreen: FunctionComponent<ExplanationScreenProps> = ({
   explanationScreenContent,
-  explanationScreenStyles,
   explanationScreenActions,
 }: ExplanationScreenProps) => {
-  useStatusBarEffect(explanationScreenStyles.statusBarStyle)
-
-  const determineIconStyle = (iconStyle: IconStyle): ViewStyle => {
-    switch (iconStyle) {
-      case IconStyle.Blue:
-        return { ...style.icon, ...style.blueIcon }
-      case IconStyle.Gold:
-        return { ...style.icon, ...style.goldIcon }
+  const { t } = useTranslation()
+  const navigation = useNavigation()
+
+  const determineCircleStyle = (circlePosition: number): ViewStyle => {
```

Comment:
This strikes me as coupling the data model and a the view layer a bit. what do you think of an api a bit more like:

```typescript
const PositionDots = (currentPosition: number, positionCount: number): FunctionComponent<P> => {
  const determineCircleStyle = (idx: number, position: number) => {
        if idx === position {
            return style.circleActive
        } else {
           return style.circleInactive
        }
   }

   return <View style={style.circles}>
         { [1, ... , positionCount].map((idx) => {
               <View style={determineCircleStyle(idx, currentPosition)} />
           }
          }
    </View>
```

Something like this allows the view component for the bottom dots to not have any knowledge of the dataModel, `explanationScreenContent.screenNumber` and both can change independently.

Reply:
Solid feedback here, I also think that this will make it easier to extend in the future and separates the logic part nicely.

Reply:
Thanks for the thoughtful feedback! Thoughts on faad0fda98b793690d80389bae588320b705130d?